avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroSerDe.java

```diff
@@ -64,19 +66,27 @@ protected void configureClientProperties(AbstractKafkaAvroSerDeConfig config) {
     }
   }
 
-  private static String validateAndMaybeGetMockScope(List<String> urls) {
-    String mockScope = null;
-    for (String url : urls) {
-      final boolean isMock = url.startsWith("mock://");
-      if (isMock && mockScope != null) {
-        throw new ConfigException(
-            "Only one mock scope is permitted for 'schema.registry.url'. Got: " + urls
-        );
-      } else {
-        mockScope = url.substring("mock://".length());
+  private static String validateAndMaybeGetMockScope(final List<String> urls) {
+    final List<String> mockScopes = new LinkedList<>();
+    for (final String url : urls) {
+      if (url.startsWith(MOCK_URL_PREFIX)) {
+        mockScopes.add(url.substring(MOCK_URL_PREFIX.length()));
       }
     }
-    return mockScope;
+
+    if (mockScopes.size() > 1) {
+      throw new ConfigException(
+              "Only one mock scope is permitted for 'schema.registry.url'. Got: " + urls
+      );
+    } else if (urls.size() > mockScopes.size()) {
```

Comment:
`mockScopes.size()` might be 0, so I think we want to move the `if (mockScopes.isEmpty())` scope before this check, so that this check will occur if we have exactly one mock

Reply:
Good eye! I just noticed this was the cause of the test failure, and did exactly what you recommended! :)